compiler/src/iree/compiler/Codegen/LLVMGPU/test/ROCDL/pipeline_tile_and_fuse.mlir

```diff
@@ -126,3 +126,93 @@ hal.executable public @main {
+        %5 = tensor.empty() : tensor<2x16x16x1280xf32>
+        %6 = tensor.empty() : tensor<2x256x11520xf16>
+        %7 = iree_linalg_ext.im2col
+            strides = [2, 2] dilations = [1, 1] kernel_size = [3, 3]
+            m_offset = [0] k_offset = [0]
+            batch_pos = [0] m_pos = [1, 2] k_pos = [3]
+          ins(%3 : tensor<2x34x34x1280xf16>)
+          outs(%6 : tensor<2x256x11520xf16>) -> tensor<2x256x11520xf16>
+        %collapsed = tensor.collapse_shape %4 [[0, 1, 2], [3]] : tensor<3x3x1280x1280xf16> into tensor<11520x1280xf16>
+        %collapsed_0 = tensor.collapse_shape %5 [[0], [1, 2], [3]] : tensor<2x16x16x1280xf32> into tensor<2x256x1280xf32>
+        %8 = linalg.fill ins(%cst : f32) outs(%collapsed_0 : tensor<2x256x1280xf32>) -> tensor<2x256x1280xf32>
+        %9 = linalg.generic {
+          indexing_maps = [affine_map<(d0, d1, d2, d3) -> (d0, d1, d3)>,
+                           affine_map<(d0, d1, d2, d3) -> (d3, d2)>,
+                           affine_map<(d0, d1, d2, d3) -> (d0, d1, d2)>],
+          iterator_types = ["parallel", "parallel", "parallel", "reduction"]}
+          ins(%7, %collapsed : tensor<2x256x11520xf16>, tensor<11520x1280xf16>)
+          outs(%8 : tensor<2x256x1280xf32>) attrs =  {lowering_config = #config} {
+        ^bb0(%in: f16, %in_1: f16, %out: f32):
+          %10 = arith.extf %in : f16 to f32
+          %11 = arith.extf %in_1 : f16 to f32
+          %12 = arith.mulf %10, %11 : f32
+          %13 = arith.addf %12, %out : f32
+          linalg.yield %13 : f32
+        } -> tensor<2x256x1280xf32>
+        %expanded = tensor.expand_shape %9 [[0], [1, 2], [3]] output_shape [2, 16, 16, 1280] : tensor<2x256x1280xf32> into tensor<2x16x16x1280xf32>
+        flow.dispatch.tensor.store %expanded, %2, offsets = [0, 0, 0, 0], sizes = [2, 16, 16, 1280], strides = [1, 1, 1, 1] : tensor<2x16x16x1280xf32> -> !flow.dispatch.tensor<writeonly:tensor<2x16x16x1280xf32>>
+        return
+      }
+    }
+  }
+}
+
+// CHECK-LABEL: func @conv_igemm_im2col
+//   CHECK-DAG:   %[[B0:.+]] = hal.interface.binding.subspan set(0) binding(0)
+//   CHECK-DAG:   %[[B1:.+]] = hal.interface.binding.subspan set(0) binding(1)
+//   CHECK-DAG:   %[[B2:.+]] = hal.interface.binding.subspan set(0) binding(2)
+//   CHECK-DAG:   memref.alloc() : memref<1x64x32xf16, #gpu.address_space<workgroup>>
+//   CHECK-DAG:   memref.alloc() : memref<32x64xf16, #gpu.address_space<workgroup>>
+//   CHECK-DAG:   %[[C0:.+]] = arith.constant 0 : index
+//   CHECK-DAG:   %[[C720:.+]] = arith.constant 720 : index
+//   CHECK-DAG:   %[[C2:.+]] = arith.constant 2 : index
+//       CHECK:   %[[LOOP:.+]] = scf.for %[[IV:.+]] = %[[C0]] to %[[C720]] step %[[C2]] {{.*}} -> (vector<1x2x2x4x1xf32>)
+//       CHECK:     gpu.barrier
+//       CHECK:     %[[LHS_RD:.+]] = vector.transfer_read %[[B0]]{{.*}} vector<8xf16>
+//       CHECK:     vector.transfer_write %[[LHS_RD]]
```

Comment:
This part is from im2col op, right?

Reply:
Yes, it should ultimately look the same as the normal matmul pipeline with different indexing on the LHS (im2col side). I don't bother checking the indexing here because the maps will likely get jumbled through canonicalizations and loop fusions throughout the pipeline.